packages/tables/src/Actions/AssociateAction.php

```diff
@@ -75,21 +75,24 @@ protected function setUp(): void
             $relationship = Relation::noConstraints(fn () => $table->getRelationship());
 
             $record = $relationship->getQuery()->find($data['recordId']);
+            $records = $this->isMultiple ? $record : collect([$record]);
 
-            if ($record instanceof Model) {
-                $this->record($record);
-            }
-
-            /** @var BelongsTo $inverseRelationship */
-            $inverseRelationship = $table->getInverseRelationshipFor($record);
+            $records->each(function ($record) use ($table, $relationship) {
```

Comment:
Can we replace this with a simple `foreach ($this->isMultiple ? $record : [$record]) {` please? Then we don't need to bother with the extra closure function and passing in the vars from the parent scope?

Reply:
Sure. Done.